---
/steinicp/SteinMICP:
   ros__parameters:
      estimator: "KF"  # ICP, KF
      class_type: "SteinMICP"
      dataset_duration: 472031817139.  #1616710393017. 472031817139
      SteinICP_parameters:
           optimizer : "Adam"   # only used for original Stein ICP
           iterations: 100
           batch_size: 1000
           particle_size: 30
           normalize: false
           lr: 1. # 0.03
           max_dist: 3.0
           knn_count: 100
           frame_gap_seconds: 0.05
           using_EarlyStop: true
           convergence_steps: 3
           convergence_threshold: 0.0005
           deskew_cloud: false
           max_range: 100.0
           SVNFullGrad: false

           min_range: 8.    # 0.3 for New college cloister, 5.0 for others 10 m for waterways
           voxel_size: 1.
           map_voxel_max_points: 20
           map_voxel_size: 1.
           map_range: 300.0
           USE_Segmentation: false

           use_weight_mean: false

           save_particles: true
           pub_cloud: true
           cov_filter_type: "none"   # none, mean or "max_sliding_window"

      lidar_param:
           topic: "/velodyne_points"

      imu_param:
           topic: "/imu/data"

           rot_rw: [ 2., 2., 10. ]   #[ 1., 1., 1. ]          # [deg/sqrt(hr)]
           vel_rw: [ 50., 50., 50. ]   #[100., 100., 100. ]          # [m/s/sqrt(hr)]
           gbstd: [ 20.0, 20.0, 20.0 ]        # [deg/hr]
           abstd: [ 500.0, 500.0, 500.0 ]     # [mGal]
           corrtime: 1.0                    # [hr]

      lio_param:
           #icp_cov_scales: [5., 5., 10., 5., 5., 1.]  #[50., 50., 100., 200., 200., 50.]
           icp_cov_scales: [1., 1., 1., 1., 1., 1.]  #[50., 50., 100., 200., 200., 50.]
           const_LidarCov: [1e-4, 1e-5]
           use_constCov: false
           initposstd: [ 0.05, 0.05, 0.05 ]
           # initial velocity std, north, east and down speed in n-frame. [m/s, m/s, m/s]
           initvelstd: [ 0.05, 0.05, 0.05 ]
           # initial attitude std, roll, pitch and yaw std. [deg, deg, deg]
           initattstd: [ 0.1, 0.1, 0.1 ]

           # sensor calibration parameter transformation from imu frame to lidar frame
           extrinsic_t: [ 0.0961, -0.1338, 0.3032 ]
           extrinsic_R: [0.9999,    0.0049,    0.0153,
                         -0.0050,    1.0000,    0.0021,
                         -0.0153,   -0.0022,    0.9999,]





